core/src/test/java/org/elasticsearch/cluster/routing/allocation/AddIncrementallyTests.java

```diff
@@ -388,15 +388,21 @@ private ClusterState removeNodes(ClusterState clusterState, AllocationService se
         DiscoveryNodes.Builder nodes = DiscoveryNodes.builder(clusterState.nodes());
         ArrayList<DiscoveryNode> discoveryNodes = CollectionUtils.iterableAsArrayList(clusterState.nodes());
         Collections.shuffle(discoveryNodes, random());
+        boolean removed = false;
         for (DiscoveryNode node : discoveryNodes) {
             nodes.remove(node.getId());
+            removed = true;
             numNodes--;
             if (numNodes <= 0) {
                 break;
             }
         }
 
         clusterState = ClusterState.builder(clusterState).nodes(nodes.build()).build();
+        if (removed) {
```

Comment:
instead of introducing boolean flag, maybe just `discoveryNodes.isEmpty() == false`